Mathlib/Algebra/Algebra/Tower.lean

```diff
@@ -353,7 +349,7 @@ theorem span_algebraMap_image (a : Set R) :
 #align submodule.span_algebra_map_image Submodule.span_algebraMap_image
 
 theorem span_algebraMap_image_of_tower {S T : Type*} [CommSemiring S] [Semiring T] [Module R S]
-    [IsScalarTower R S S] [Algebra R T] [Algebra S T] [IsScalarTower R S T] (a : Set S) :
+    [Algebra R T] [Algebra S T] [IsScalarTower R S T] (a : Set S) :
```

Comment:
This can't be generalized yet with what we include in this PR; this is an application of the `CompatibleSMul` class but there's some unresolved [controversy](https://github.com/leanprover-community/mathlib4/pull/9046#discussion_r1429119479) with @eric-wieser.